google/cloud/security/scanner/scanner.py

```diff
@@ -84,60 +121,112 @@ def main(_):
                      'Use "forseti_scanner --helpfull" for help.'))
         sys.exit(1)
 
-    rules_engine = IamRulesEngine(rules_file_path=FLAGS.rules)
+    # Instantiate rules engine with supplied rules file
+    rules_engine = rules_engine_class(rules_file_path=FLAGS.rules)
     rules_engine.build_rule_book()
 
     snapshot_timestamp = _get_timestamp()
     if not snapshot_timestamp:
         LOGGER.warn('No snapshot timestamp found. Exiting.')
         sys.exit()
 
-    # TODO: make this generic
-    org_policies = _get_org_policies(snapshot_timestamp)
-    project_policies = _get_project_policies(snapshot_timestamp)
+    # Load pipeline from mapping file
+    engine_map = pc.get_engine_info(rules_engine_name)
 
-    if not org_policies and not project_policies:
-        LOGGER.warn('No policies found. Exiting.')
-        sys.exit()
+    data_pipeline_class, _ = _create_rules_engine(
```

Comment:
As noted above, perhaps we can create a map and not need the  ```_create_rules_engine()```?

For example, we can do something like this:
```python

scanner_map  = {
    'IamPolicy': scanners.IamPolicyScanner,
    'Foo': scanners.FooScanner,
    'Bar':  scanners.BarScanner,
  }

# So, that we can do something like this:
scanner = scanner_map['IamPolicy'](snapshot_timestamp)
```